test/test_SW_VegProd.cc

```diff
@@ -159,4 +159,999 @@ namespace {
+             Trees_Fraction = 0.0, fix_trees = TRUE,
+             SumGrasses_Fraction = .5, fix_sumgrasses = TRUE,
+             BareGround_Fraction = .5, fix_BareGround = TRUE,
+             dailyC4vars = clim1[["dailyC4vars"]],
+             fix_issue218 = TRUE, fix_issue219 = TRUE
+           )
+         ```
+         */
+
+        inputValues[succIndex] = 0.;
+        inputValues[forbIndex] = 0.;
+        inputValues[C3Index] = SW_MISSING;
+        inputValues[C4Index] = SW_MISSING;
+        inputValues[grassAnn] = 0.;
+        inputValues[shrubIndex] = 0.;
+        inputValues[treeIndex] = 0.;
+        inputValues[bareGround] = .5;
+
+        SumGrassesFraction = .5;
+
+        RelAbundanceL0Expected[succIndex] = 0.;
+        RelAbundanceL0Expected[forbIndex] = 0.;
+        RelAbundanceL0Expected[C3Index] = .5;
+        RelAbundanceL0Expected[C4Index] = 0.;
+        RelAbundanceL0Expected[grassAnn] = 0.;
+        RelAbundanceL0Expected[shrubIndex] = 0.;
+        RelAbundanceL0Expected[treeIndex] = 0.;
+        RelAbundanceL0Expected[bareGround] = .5;
+
+        RelAbundanceL1Expected[treeIndexL1] = 0.;
+        RelAbundanceL1Expected[shrubIndexL1] = 0.;
+        RelAbundanceL1Expected[forbIndexL1] = 0.; // Constains forbs + succulents (L0)
+        RelAbundanceL1Expected[grassesIndexL1] = .5;
+        RelAbundanceL1Expected[bareGroundL1] = .5;
+
+        estimatePotNatVegComposition(climateAverages.meanTemp_C, climateAverages.PPT_cm,
+            climateAverages.meanTempMon_C, climateAverages.PPTMon_cm, inputValues, shrubLimit,
+            SumGrassesFraction, C4Variables, fillEmptyWithBareGround, inNorthHem, warnExtrapolation,
+            fixBareGround, grassOutput, RelAbundanceL0, RelAbundanceL1);
+
+        // Loop through RelAbundanceL0 and test results.
+        for(index = 0; index < nTypes; index++) {
+            EXPECT_DOUBLE_EQ(RelAbundanceL0[index], RelAbundanceL0Expected[index]);
+        }
+
+        // Loop through RelAbundanceL1 and test results
+        for(index = 0; index < 5; index++) {
+            if(index != grassesIndexL1) {
+                EXPECT_DOUBLE_EQ(RelAbundanceL1[index], RelAbundanceL1Expected[index]);
+            } else {
+                EXPECT_NEAR(RelAbundanceL1[grassesIndexL1], SumGrassesFraction, tol6);
+            }
+        }
+
+        // Deallocate structs
+        allocDeallocClimateStructs(deallocate, 31, &climateOutput, &climateAverages);
+
+    }
+
+    TEST(VegEstimationDeathTest, VegInputGreaterThanOne) {
+
+        /*  ================================================================
+                   Tests a death case of `estimatePotNatVegComposition()`
+                        when input vegetation values sum to over 1
+            ================================================================  */
+
+        SW_CLIMATE_CLIM climateAverages;
+        SW_CLIMATE_YEARLY climateOutput;
+
+        double SumGrassesFraction = SW_MISSING;

```

Comment:
Please add a new death test that covers the situation of https://github.com/DrylandEcology/rSOILWAT2/issues/219